dotcom-rendering/scripts/webpack/gu-stats-report-plugin.js

```diff
@@ -0,0 +1,158 @@
+const fetch = require('node-fetch');
+const os = require('os');
+const { exec } = require('child_process');
+
+class GuStatsReportPlugin {
+	constructor(config) {
+		this.buildName = config?.buildName;
+		this.project = config?.project;
+		this.team = config?.team;
+		this.buildCount = 0;
+		this.gitBranch = null;
+		this.gitHash = null;
+		this.machineUsername = null;
+		this.machineHostname = os.hostname();
+		this.fetchMachineUsername();
+		this.fetchGitBranch();
+		this.fetchGitHash();
+		if (config.displayDisclaimer)
+			console.log(
+				'[gu-stats-report] This project reports build stats & basic machine information for internal use only. We will use this information to help us improve developer experience for this project.',
+			);
+	}
+
+	isValidConfig() {
+		if (!this.buildName || !this.project || !this.team) return false;
+		return true;
+	}
+
+	fetchMachineUsername() {
+		try {
+			// if 'username' is undefied os.userInfo() will throw an error
+			// https://nodejs.org/api/os.html#osuserinfooptions
+			this.machineUsername = os.userInfo().username;
+		} catch {
+			console.error('[gu-stats-report] Failed to get machine username');
+		}
+	}
+
+	fetchGitBranch() {
+		exec('git branch --show-current', (err, stdout) => {
+			if (err)
+				return console.error(
+					'[gu-stats-report] Failed to get current git branch',
+				);
+			this.gitBranch = stdout.trim();
+		});
+	}
+
+	fetchGitHash() {
+		exec('git rev-parse --short HEAD', (err, stdout) => {
+			if (err)
+				return console.error(
+					'[gu-stats-report] Failed to get current git hash',
+				);
+			this.gitHash = stdout.trim();
+		});
+	}
+
+	apply(compiler) {
+		const onDone = (stats) => {
+			// Increment the buildCount
+			this.buildCount += 1;
+
+			if (!this.isValidConfig)
+				return console.log(
+					'[gu-stats-report] Unable to report stats - invalid config',
+				);
+
+			const URL = 'https://logs.code.dev-guardianapis.com/log';
+			fetch(URL, {
+				method: 'POST',
+				body: JSON.stringify({
+					label: 'buildstats',
```

Comment:
Ping @sndrs FYI there was a decision to have this as a generic `label` so that we could potentially have a generic dashboard for other CS build systems.